### DISCUSSION TITLE
Is there a way to destroy a zpool created on a node?

### DISCUSSION THREAD

**Discussion by talosnoob** [siderolabs/talos#9928] [Q&A] [answered]
Hello Talos team. 

I have a test cluster that build on virtual machines.  

I installed and configure zfs zpool on worker node according to this document: 

https://github.com/siderolabs/extensions/blob/main/storage/zfs/README.md

So now I have a zfs pool on node: 

```
# talosctl mounts | grep "/var/my-zfs-pool"
192.168.50.21   my-zfs-pool   51.51      0.00       51.51           0.00%          /var/my-zfs-pool

``` 

But when I tried to destroy this pool, I get a message: 

"cannot destroy 'my-zfs-pool': pool is busy"

```
/ # chroot /host zpool destroy -f my-zfs-pool
cannot destroy 'my-zfs-pool': pool is busy

``` 

Is there a way to destroy a zpool created on a node?

**Comment by smira** [comment, marked as answer]
The zfs extension is a community extension, so we can't offer support at the moment.

I guess you can't destroy a mounted ZFS pool if that makes sense?

**Comment by talosnoob** [comment]
If I'm not wrong, pool currently not mounted but I can't destroy it:

```
/ # chroot /host zpool list
NAME          SIZE  ALLOC   FREE  CKPOINT  EXPANDSZ   FRAG    CAP  DEDUP    HEALTH  ALTROOT
my-zfs-pool  49.5G  1.03M  49.5G        -         -     0%     0%  1.00x    ONLINE  -

/ # chroot /host zfs umount -f my-zfs-pool
cannot unmount 'my-zfs-pool': not currently mounted

/ # chroot /host zpool destroy my-zfs-pool
cannot destroy 'my-zfs-pool': pool is busy
/ # 

``` 

Ok,  I'll try to google again, may be more lucky this time.  Thanks.
